### PR TITLE
Recipe Audit 1.18

### LINKED ISSUE #8213: [BUG] Stone mason does not want to craft Chiseled Nether Brick
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

minecolonies-1.18.1-1.0.511-BETA

### Structurize Version

structurize-1.18.1-1.0.352-ALPHA

### Related Mods and their Versions

_No response_

### Current Behavior

Stone mason does not accept recipe

### Expected Behavior

Stone mason does accept recipe

### Reproduction Steps

1. Teach Chiseled Nether Brick recipe to stone mason
2. Recive "Unable to store" -Recipe Message

### Logs

https://gist.github.com/AlmightyAnubis/16ec28cd76962b8093d0c0f47c36ec6c

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8224: [BUG] Stone Mason does not save Custom Recipes using colored Terracotta blocks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

1.18.2-1.0.568-BETA

### Structurize Version

1.18.2-1.0.370-ALPHA

### Related Mods and their Versions

_No response_

### Current Behavior

I go to the Custom Recipe tab on the Stonemason hut to enter a recipe using a colored Terracotta block (for example to make White Terracotta Wall as requested for a Lost City construction). When I click on the teach button it says that it cannot save either because there are no spaces left or it isn't appropriate for that worker. 
It is only the 3rd recipe I've taught it. The plain Terracotta block is accepted as a valid recipe.

### Expected Behavior

The recipe would be learned successfully.

### Reproduction Steps

1. Go to Stonemason hut
2. choose Custom Recipe tab
3. under first input choose any colored Terracotta block
4. choose Teach

### Logs

If you want logs let me know.

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by uecasm**
Closes #8213
Closes #8224

# Changes proposed in this pull request:
- Add recipe auditor
    - When enabled, writes some CSV files indicating current colonist recipe crafting and tags coverage, on world startup and /reload
    - Should be especially useful after an MC version upgrade or when building a modpack
    - Only shows crafting; some other sources of items like fishing and animal huts are not counted
- Fixes some lingering issues where JEI was not showing some recipes that were actually teachable.
- Updates various tags to handle most missing items
- Misc other files touched by runData

Review please

See [this spreadsheet](https://docs.google.com/spreadsheets/d/1fPdA5BDmnW3r2BXfRik8bmVt0nxwI3tOrrpc1vYXP_I/edit#gid=573731263) for a before/after report (generated by the auditor).

### CODE REVIEW

File: src/main/resources/assets/minecolonies/lang/manual_en_us.json [hunk 1]
```diff
@@ -241,6 +241,8 @@
   "minecolonies.config.fishingroddurabilityadjustt2.comment": "Offset for the maximum durability unenchanted rod a T2 Fishing Hut can use, compared to iron tools (250). Thermal Foundation Iron requires +6. T3 huts can use unenchanted rods of any durability level, so long as they can take damage at all. Anything below -250 will only allow vanilla rods, except as allowed by T1.",
   "minecolonies.config.enabledmodtags": "Enabled Mod Tags",
   "minecolonies.config.enabledmodtags.comment": "List of recognized and enabled tags.",
+  "minecolonies.config.auditcraftingtags": "Audit Crafting Tags",
+  "minecolonies.config.auditcraftingtags.comment": "Generate crafting_audit.csv on loading recipes to verify tags.",
```
Comment (someaddons) [inline src/main/resources/assets/minecolonies/lang/manual_en_us.json:245]: bit weak of a desc for the users, maybe "If enabled crafting dumps crafting item and citizen crafting recipes into crafting_audit.csv on loading"

File: src/datagen/generated/minecolonies/data/minecolonies/tags/items/dyed_terracotta.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "minecraft:pink_terracotta",
+    "minecraft:gray_terracotta",
+    "minecraft:light_gray_terracotta",
+    "minecraft:cyan_terracotta",
+    "minecraft:purple_terracotta",
+    "minecraft:blue_terracotta",
+    "minecraft:brown_terracotta",
+    "minecraft:green_terracotta",
+    "minecraft:red_terracotta",
+    "minecraft:black_terracotta"
```
Comment (Raycoms) [inline src/datagen/generated/minecolonies/data/minecolonies/tags/items/dyed_terracotta.json]: I think DO has a tag for terracotta

Comment (uecasm) [reply src/datagen/generated/minecolonies/data/minecolonies/tags/items/dyed_terracotta.json]: Actually it looks like vanilla MC does in 1.18... they didn't in 1.16 though.  I'll change that over.

File: src/main/resources/assets/minecolonies/lang/manual_en_us.json (hunk 1 above)
Comment (uecasm) [reply src/main/resources/assets/minecolonies/lang/manual_en_us.json:245]: I don't think that's any clearer than what I already wrote.

Comment (someaddons) [reply src/main/resources/assets/minecolonies/lang/manual_en_us.json:245]: well then come up with something else better :)

**Review by Raycoms — APPROVED** [review]

**Pull request merged**